source/jormungandr/jormungandr/parking_space_availability/car/star.py

```diff
@@ -0,0 +1,95 @@
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+#
+# Stay tuned using
+# twitter @navitia
+# IRC #navitia on freenode
+# https://groups.google.com/d/forum/navitia
+# www.navitia.io
+from __future__ import absolute_import, print_function, unicode_literals, division
+
+import logging
+import pybreaker
+import requests as requests
+import jmespath
+
+from jormungandr import cache, app
+from jormungandr.parking_space_availability import AbstrcatParkingPlacesProvider
+from jormungandr.parking_space_availability.car.parking_places import ParkingPlaces
+from jormungandr.ptref import FeedPublisher
+
+DEFAULT_STAR_FEED_PUBLISHER = None
+
+
+class StarProvider(AbstrcatParkingPlacesProvider):
+
+    WS_URL_TEMPLATE = 'http://data.explore.star.fr/api/records/1.0/search/?dataset={}&refine.idparc={}'
+
+    def __init__(self, operators, timeout, dataset, feed_publisher=DEFAULT_STAR_FEED_PUBLISHER):
+        self.operators = [o.lower() for o in operators]
+        self.timeout = timeout
+        self.dataset = dataset
+        self.breaker = pybreaker.CircuitBreaker(fail_max=5, reset_timeout=10)
+        self._feed_publisher = FeedPublisher(**feed_publisher) if feed_publisher else None
+
+    def support_poi(self, poi):
+        properties = poi.get('properties', {})
+        return properties.get('operator', '').lower() in self.operators
+
+    def get_informations(self, poi):
+        ref = poi.get('properties', {}).get('ref')
+        if not ref:
+            return
+
+        data = self._call_webservice(ref)
+        if not data:
+            return
+
+        available_places = jmespath.search('records[0].fields.nombreplacesdisponibles', data) or 0
+        occupied_places = jmespath.search('records[0].fields.nombreplacesoccupees', data) or 0
+        available_disabled = jmespath.search('records[0].fields.nombreplacesdisponiblespmr', data) or 0
+        occupied_disabled = jmespath.search('records[0].fields.nombreplacesoccupeespmr', data) or 0
+
+        return ParkingPlaces(available_places, occupied_places, available_disabled, occupied_disabled)
+
+    @cache.memoize(app.config['CACHE_CONFIGURATION'].get('TIMEOUT_STAR', 30))
+    def _call_webservice(self, parking_id):
+        try:
+            data = self.breaker.call(requests.get, self.WS_URL_TEMPLATE.format(self.dataset, parking_id),
+                                     timeout=self.timeout)
+            return data.json()
+        except pybreaker.CircuitBreakerError as e:
+            logging.getLogger(__name__).error('STAR service dead (error: {})'.format(e))
```

Comment:
You should take advantage of the new json logging mechanize.
You can look at timeo or siri_lite for some example